File: src/Level1/이상한문자만들기.java
package Level1;

public class 이상한문자만들기 {
    public String solution(String s) {
        String answer="";
        StringBuilder sb = new StringBuilder();

        int j = 1;
        for(int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != ' ') {
                sb.append(j % 2 != 0 ? Character.toUpperCase(s.charAt(i)) : Character.toLowerCase(s.charAt(i)));
                j++;
            }
            else {
                sb.append(s.charAt(i));
                j = 1;
            }
        }

        answer = sb.toString();
        return answer;
    }
}

File: src/Level1/이상한문자만들기.kt
package Level1

fun main(){
    print(solution("try hello world"))
    print(solution1("try hello world"))
}

private fun solution(s: String): String{
    var answer = ""
    var wi = 0
    s.forEach{
        answer += if(wi % 2 == 0) it.toUpperCase()
        else it.toLowerCase()
        wi++
        if(it == ' ') wi = 0
    }
    return answer
} // 이게 시간복잡도가 덜할 듯

private fun solution1(s: String): String{
    var answer = ""
    s.split(" ").forEach{
        for(i in 0 until it.length){
            answer += if(i % 2 == 0) it[i].toUpperCase()
            else it[i].toLowerCase()
        }
        answer += " "
    }
    return answer
}